### PR TITLE
Fix input resource leak

### PR DESCRIPTION
**Pull request by ssoloff**
## Overview

As identified by LGTM.  Fixes an `InputStream` resource leak in `HelpSupport#loadHelp()`.

## Functional Changes

* Close the `InputStream` from which the help file is read.
* Check for a `null` return value from `Class#getResourceAsStream()` to avoid a potential NPE.

## Refactoring Changes

* Use `IOUtils#toString()` instead of reading the stream manually.
* Add missing Javadocs.

## Manual Testing Performed

* Verified help text for PBEM/PBF providers is displayed correctly.
* Manually triggered a missing help file error to confirm the error dialog is displayed and the appropriate error message is displayed in the HTML help window.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/help/HelpSupport.java [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
-      final InputStream is = HelpSupport.class.getResourceAsStream(fileName);
-      final BufferedReader br = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
-      String line;
-      final StringBuilder sb = new StringBuilder();
-      while ((line = br.readLine()) != null) {
-        sb.append(line);
+    checkNotNull(fileName);
+
+    try (@Nullable
+    InputStream is = HelpSupport.class.getResourceAsStream(fileName)) {
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/triplea/help/HelpSupport.java]: An unfortunate consequence of the Eclipse formatter.

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/triplea/help/HelpSupport.java]: I didn't even know try with resource allows null in the sense that it doesn't throw an exception when null is used.

Comment (ssoloff) [reply game-core/src/main/java/games/strategy/triplea/help/HelpSupport.java]: Yeah, if you look at the generated bytecode, you'll see that it checks to see if the resource is `null` in the equivalent of the `finally` block before attempting to close the resource.

**Comment by ron-murhammer** [comment]
This pull request **fixes 1 alert** when merging 5b3ecd449cd946516c3005300e8a0c3477993010 into 7a9166501cd4ae1ee6e184f7bc6db977f6e7eb52 - [view on LGTM.com](https://lgtm.com/projects/g/triplea-game/triplea/rev/pr-b443662116275d7e5c71ce8acca70da943e30a71)

**fixed alerts:**

* 1 for Potential input resource leak

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by RoiEXLab** [comment]
The notification of fixed issues is nice, however you might want to change it so the @tripleabuilderbot posts those notifications instead @ron-murhammer 😅

**Pull request merged**